core/src/main/java/org/bitcoinj/script/Script.java

```diff
@@ -93,6 +94,7 @@
 
     // The program is a set of chunks where each element is either [opcode] or [data, data, data ...]
     protected List<ScriptChunk> chunks;
+    private ScriptPattern pattern;
```

Comment:
Can this be final?

Reply:
What's the reason you are storing the pattern? Am I missing some use-case?
I know anyone would define its - no bitcoin specific - patterns, but here `pattern` is used to answer methods like `isPayToScriptHash`, which are bitcoin specific.

Reply:
It is useful if you wanted to add a modified ruleset but in the case of the Bitcoin specific `isPayToScriptHash` you could return false.

Reply:
@natzei Are you fine with this? I'd merge if you think your comment has been addressed.

Reply:
Sorry for the delay.
@erasmospunk You didn't convinced me. After you added a modified ruleset within a Script, what's the purpose? The method `chunksMatch` you wrote is actually fine.
At the time, I don't see the necessity to store the pattern. Maybe I'm wrong, btw.